CONTRIBUTING.md

```diff
@@ -345,6 +345,18 @@ with clear instructions on how to install the required libraries.
 Furthermore, if your package requires CGO, it MUST be able to compile and operate in a no op mode
 or report a warning back to the collector with a clear error saying CGO is required to work.
 
+### Breaking changes
+
+Whenever possible, we adhere to semver as our minimum standards. Even before v1, we strive to not break compatibility
+without a good reason. Hence, when a change is known to cause a breaking change, it should be clearly marked in the
+changelog, possibly with a line instructing users how to move forward.
+
+We also strive to perform breaking changes in two stages, or using feature flags: 
+
+- when we need to remove something, we mark a feature as deprecated in one version, and remove it in the next one
+- when renaming types, functions, or attributes, we create the new name first, deprecating the old one in one
+version, removing it in the next version
+- when creating new interfaces, provide helpers so that consumers can be forward compatible without much effort
```

Comment:
Not sure if we want to commit to this. Maybe as an "optional"? The helper will be remove anyway next release, so the dependency has to change.

Reply:
Do we have examples of such helpers?  It's not clear to me what this would look like.

Reply:
I had the configauth example in mind, where we have a ServerAuthenticator interface, as well as a `NewServerAuthenticator` function that returns an implementation based on the provided options.

https://github.com/open-telemetry/opentelemetry-collector/blob/6b5a3d08a96bfb41a5e121b34f592a1d5c6e0435/config/configauth/default_serverauthenticator.go#L59-L71

Reply:
I added something about it to the doc.